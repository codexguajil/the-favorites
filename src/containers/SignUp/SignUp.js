import React, { Component } from 'react';
import { fetchData } from '../../utils/fetch';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }
  
  handleAddUser = async (e) => {
    e.preventDefault();
    const url = 'http://localhost:3000/api/users/new#'
    fetchData(url, {
      method: 'POST',
      body: JSON.stringify(this.state),
      headers: {
        'Content-Type': 'application/json'
      }
    })
  }

  render() {
    return (
      <div className="logout">
        <form onSubmit={this.handleSignIn} className="form">
        <Link to={'/'} className='back-btn'>Back To Home</Link>
        <input
            value={this.state.name}
            name='name'
            placeholder="Your Name"
            onChange={this.handleChange}
            className="input"
          />
          <input
            value={this.state.email}
            name='email'
            placeholder="Email"
            onChange={this.handleChange}
            className="input"
          />
          <input
            value={this.state.password}
            type='password'
            name='password'
            placeholder="Password"
            onChange={this.handleChange}
            className="input"
          />
          <button onClick={this.handleAddUser} className="sign-up-btn">Sign Up</button>
        </form>
      </div>
    )
  }
}

export default SignUp;