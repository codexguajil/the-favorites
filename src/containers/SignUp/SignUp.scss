.sign-up-btn {
  background: hsla(0, 0%,10%, 1);
  border: none;
  border-radius: 5px;
  color: hsla(360, 0%, 100%, 1);
  font-family: 'bebas';
  font-size: 32px;
  height: 40px;
  letter-spacing: .2em;
  margin: auto;
  width: 95%;
  &:hover {
    background: hsla(360, 0%, 100%, 1);
    color: hsla(0, 0%,10%, 1);
  }
}