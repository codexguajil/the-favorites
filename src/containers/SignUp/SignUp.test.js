import React from 'react';
import SignUp from './SignUp'
import { shallow } from 'enzyme'

describe('SignUp', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <SignUp />
    )

    expect(wrapper).toMatchSnapshot()
  })
})