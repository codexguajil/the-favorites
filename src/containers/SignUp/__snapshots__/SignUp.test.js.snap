// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `
<div>
  <Link
    className="back-btn"
    to="/"
  >
    Back To Home
  </Link>
  <form>
    <input
      name="name"
      onChange={[Function]}
      placeholder="your name"
      value=""
    />
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Sign Up
    </button>
  </form>
</div>
`;
